File: LC_605.py
from typing import *

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        
        #[1, 0, 0, 0, 1, 0, 1]

        for i in range(len(flowerbed)):

            if n == 0:
                return True

            if i == 0 and flowerbed[i] == 0 and len(flowerbed) != 1 and flowerbed[i+1] == 0:
                flowerbed[i] = 1
                n -= 1
            elif i == len(flowerbed)-1 and flowerbed[i - 1] == 0 and flowerbed[i] == 0:
                flowerbed[i] = 1
                n -= 1
            else:
                if i != 0 and i != len(flowerbed)-1 and flowerbed[i-1] == 0 and flowerbed[i+1] == 0 and flowerbed[i] == 0:
                    flowerbed[i] = 1
                    n -= 1
                
        return True if n == 0 else False
    
print(Solution().canPlaceFlowers([1,0,0,0,1], 2))

File: LC_605.java
class Solution {
    public static boolean canPlaceFlowers(int[] flowerbed, int n) {
        
        if (flowerbed.length == 1) {
    if (flowerbed[0] == 0) n--;
    return n <= 0;
}

        for(int i = 0; i < flowerbed.length; i++)
        {
            if(n == 0)
            return true;

            if(i == 0 && flowerbed.length != 1) {
                if(flowerbed[i] == 0 && flowerbed[i+1] == 0) {
                    flowerbed[i] = 1;
                    n--;
                }
            }
            else if( i == flowerbed.length - 1 && flowerbed.length != 1) {
                if(flowerbed[i] == 0 && flowerbed[i-1] == 0) {
                    flowerbed[i] = 1;
                    n--;
                }
            }
            else {
                if(flowerbed[i-1] == 0 && flowerbed[i+1] == 0 && flowerbed[i] == 0) {
                    flowerbed[i] = 1;
                    n--;
                }
            }
        }

        return n==0;
    }

    public static void main(String[] args) {
        System.out.println(Solution.canPlaceFlowers(new int[]{0}, 1));
    }
}